src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs

```diff
@@ -102,6 +103,9 @@ public static void AddWebJobsScriptHost(this IServiceCollection services, IConfi
             services.AddTransient<VirtualFileSystem>();
             services.AddTransient<VirtualFileSystemMiddleware>();
 
+            // Logging and diagnostics
+            services.AddSingleton<IMetricsLogger, WebHostMetricsLogger>();
```

Comment:
Was the metrics logger not available for host-level services?

Reply:
yes, it's available for host-level services. I sent another commit.